### ISSUE TITLE
Deep learning based clustering for detectors

### ISSUE THREAD

**Issue by jpata** [HEP-KBFI/student-projects#6]
So far, we can do the particle reconstruction step (c) -> (d) with ML, see e.g. https://www.nature.com/articles/s42005-024-01599-5.

<img width="1750" height="822" alt="Image" src="https://github.com/user-attachments/assets/0b2a6353-472f-4a00-94f3-3d26260075e7" />

Now we also want to integrate the calorimeter clustering step (b) -> (c) to the ML model.

There are no straightforward existing models that can be applied out of the box, but some general approaches can be found in the literature:

- https://arxiv.org/pdf/2007.03083
- https://arxiv.org/pdf/2102.01033
- https://indico.cern.ch/event/1156222/contributions/5062801/attachments/2521271/4336402/2022%20FastML%20Workshop%20--%20SPVCNN%20(Final).pdf
- https://indico.cern.ch/event/1283970/contributions/5554353/attachments/2720694/4726921/JK_SPVCNN_FastML.pdf 
- https://indico.cern.ch/event/1046620/contributions/4397048/attachments/2267469/3851829/SPVCNN%20-%20EPE%20Data%20Science%20(2021-06-18).pdf

First steps:
- For a very basic intro, use `sklearn.datasets.make_blobs` to make a toy dataset of clusters and their labels
- build and optimize a simple deep learning based clustering model on this toy dataset, compare the results to some sklearn clustering model
- The real datasets are defined in https://github.com/erwulff/particlemind/blob/main/notebooks/debug-cld-processing.ipynb, a few files can be downloaded from https://jpata.web.cern.ch/jpata/mlpf/cld_key4hep/reco_p8_ee_tt_ecm365/parquet/
